be/src/exec/pipeline/sink/memory_scratch_sink_operator.cpp

```diff
@@ -0,0 +1,120 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+#include "exec/pipeline/sink/memory_scratch_sink_operator.h"
+
+#include "util/arrow/row_batch.h"
+#include "util/arrow/starrocks_column_to_arrow.h"
+
+namespace starrocks::pipeline {
+
+Status MemoryScratchSinkOperator::prepare(RuntimeState* state) {
+    RETURN_IF_ERROR(Operator::prepare(state));
+    return Status::OK();
+}
+
+void MemoryScratchSinkOperator::close(RuntimeState* state) {
+    Operator::close(state);
+}
+
+bool MemoryScratchSinkOperator::need_input() const {
+    if (_pending_result == nullptr) {
+        return true;
+    }
+    if (_queue->try_put(_pending_result)) {
+        _pending_result.reset();
+        return true;
+    }
+    return false;
+}
+
+bool MemoryScratchSinkOperator::is_finished() const {
+    return _is_finished;
+}
+
+Status MemoryScratchSinkOperator::set_finishing(RuntimeState* state) {
+    _is_finished = true;
+    return Status::OK();
+}
+
+bool MemoryScratchSinkOperator::pending_finish() const {
+    const_cast<MemoryScratchSinkOperator*>(this)->try_to_put_sentinel();
+    return !(_is_finished && _pending_result == nullptr && _has_put_sentinel);
+}
+
+Status MemoryScratchSinkOperator::set_cancelled(RuntimeState* state) {
```

Comment:
add some comment to explain why MemoryScratchSinkOperator needs to override set_cancelled, at the present time, only ExchangeSink overrides this function to handle brpc errors, other operators' set_cancelled is trivial and it is equivallent to set_finished function.

Reply:
done